apacheconfig/wloader.py

```diff
@@ -8,10 +8,68 @@
 
 import abc
 import six
+import contextlib
+
+from apacheconfig.lexer import make_lexer
+from apacheconfig.parser import make_parser
+from apacheconfig.reader import LocalHostReader
 
 from apacheconfig import error
 
 
+@contextlib.contextmanager
+def make_writable_loader(**options):
+    # Preserve whitespace is necessary for the writable loader to work.
+    options['preservewhitespace'] = True
+    ApacheConfigLexer = make_lexer(**options)
+    ApacheConfigParser = make_parser(**options)
+    yield ApacheConfigWritableLoader(ApacheConfigParser(ApacheConfigLexer(),
+                                                        start='contents'),
+                                     **options)
+
+
+class ApacheConfigWritableLoader(object):
+    """Manages a writable object represented by a single config file.
+    """
+
+    def __init__(self, parser, **options):
+        self._parser = parser
+        self._options = dict(options)
+        if 'reader' in self._options:
```

Comment:
nit: arguably, `try/except` is more Pythonic

Reply:
I'm following the logic in `apacheconfig/loader.py` so that they work similarly-- do you think the constructor should fail rather than using a default Reader?

Reply:
No! I just recalled Python motto "Ask for forgiveness not permission". That is, try the thing you want to, be prepared to handle possible failure.